### PR TITLE
Add contribution guidelines

### LINKED ISSUE #16: Add contribution guidelines
It would be good to add contribution guidelines for people who might be interested in contributing? I have found the guidelines used by [geopandas](https://geopandas.org/en/stable/community/contributing.html) (which again is based on pandas) very useful.

### PR DESCRIPTION
**Pull request by HTenkanen**
Closes #16 

Added now contribution guidelines based on how they were written for pyrosm / momepy. Updated them a bit and converted the file to Markdown. The guidelines will be also visible on the website.

### CODE REVIEW

File: docs/CONTRIBUTIONS.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    conda install configargparse geopandas joblib jpype1 pandas psutil requests pytest pytest-cov codecov requests black
+
+This should install all necessary dependencies including optional and packages for running tests.
+
+### 4. Making a development build
+
+Once dependencies are in place, make an in-place build by navigating to the git
+clone of the *r5py* repository and running::
+
+    python setup.py develop
```
Comment (christophfink) [inline docs/CONTRIBUTIONS.md:107]: Does not work with pep517. 
There’s currently no exact equivalent to it, but the closest would be `pip install --user .` IIRC

Comment (christophfink) [reply docs/CONTRIBUTIONS.md:107]: or to keep it within the environment, probably a simple `pip install .`

File: docs/CONTRIBUTIONS.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+#### Python (PEP8 / black)
+
+*r5py* follows the [PEP8](http://www.python.org/dev/peps/pep-0008) standard
+and uses [Black](https://black.readthedocs.io/en/stable/) to ensure a consistent code format throughout the project.
+
+CI will run `black --check` and fails if there are files which would be
+auto-formatted by `black`. Therefore, it is helpful before submitting code to
+auto-format your code::
+
+    black r5py
```
Comment (christophfink) [inline docs/CONTRIBUTIONS.md:175]: Should be `black src`

File: docs/CONTRIBUTIONS.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+are not executed by readthedocs.
+
+### 7. Formatting the code
+
+#### Python (PEP8 / black)
+
+*r5py* follows the [PEP8](http://www.python.org/dev/peps/pep-0008) standard
+and uses [Black](https://black.readthedocs.io/en/stable/) to ensure a consistent code format throughout the project.
+
+CI will run `black --check` and fails if there are files which would be
```
Comment (christophfink) [inline docs/CONTRIBUTIONS.md:171]: Currently does not pass black, because some things become slightly less readable. Let’s remember to run it ;)

**Review by christophfink — COMMENTED** [review]
Did the conf.py and index.rst slip in from the other pull request?

File: docs/CONTRIBUTIONS.md (hunk 2 above)
Comment (HTenkanen) [reply docs/CONTRIBUTIONS.md:175]: Ah right 👍🏻

**Comment by HTenkanen** [comment]
> Did the conf.py and index.rst slip in from the other pull request?

These files are here on purpose because we want to have the contribution guidelines visible also on the website.

File: docs/CONTRIBUTIONS.md (hunk 1 above)
Comment (HTenkanen) [reply docs/CONTRIBUTIONS.md:107]: Yep, I think `pip install .` should do it. Will update accordingly

File: docs/CONTRIBUTIONS.md (hunk 3 above)
Comment (HTenkanen) [reply docs/CONTRIBUTIONS.md:171]: Yep, I think this is the goal at least :)

**Comment by christophfink** [comment]
> > Did the conf.py and index.rst slip in from the other pull request?
> 
> These files are here on purpose because we want to have the contribution guidelines visible also on the website.

Could you resolve the conflicts with the other (merged) pull request?

**Comment by HTenkanen** [comment]
> > > Did the conf.py and index.rst slip in from the other pull request?
> > 
> > 
> > These files are here on purpose because we want to have the contribution guidelines visible also on the website.
> 
> Could you resolve the conflicts with the other (merged) pull request?

Not sure what you mean here? So do you want to push these changes in files to the PR / branch that was merged earlier instead of this PR? Or?

File: docs/CONTRIBUTIONS.md (hunk 2 above)
Comment (HTenkanen) [reply docs/CONTRIBUTIONS.md]: This is now fixed.

File: docs/CONTRIBUTIONS.md (hunk 1 above)
Comment (HTenkanen) [reply docs/CONTRIBUTIONS.md]: This is fixed.

**Comment by HTenkanen** [comment]
@christophfink : Now updated the contribution guidelines. Didn't touch to the conf.py and index.rst because I'm not sure what you want to do with them, or what is the issue there. 🙂

**Comment by christophfink** [comment]
Do you want to keep nbsphinx and myst_parser in?

**Comment by HTenkanen** [comment]
> Do you want to keep nbsphinx and myst_parser in?

True, nbshphinx shouldn't be needed. I believe myst_parser is needed for parsing documents from Readme files:
https://www.sphinx-doc.org/en/master/usage/markdown.html

**Comment by HTenkanen** [comment]
@christophfink : Now I think I finally understood your comments.. Fixed the conflicts now with the conf.py.

**Review by christophfink — APPROVED** [review]
looks good

**Comment by HTenkanen** [comment]
> Do you want to keep nbsphinx and myst_parser in?

Well in the end, Sphinx didn't like the `myst_parser` extension either.. Finally got this working by excluding it. Probably `myst_nb` somehow activates the markdown support in the background..

**Pull request merged**